bootstrap_salt/fab_tasks.py

```diff
@@ -100,211 +105,121 @@ def application(application_name):
+    env.kms_data_key = create_kms_data_key()
+    bcfn_create(test=test)
 
-    Search for the salt masters zone within the project config
-    file, and then AWS if necessary, returning the fully
-    qualified domain name of the salt master
-    """
-    _validate_fabric_env()
-    project_config = config.ProjectConfig(env.config,
-                                          env.environment,
-                                          env.stack_passwords)
-    cfg = project_config.config
-    try:
-        zone_name = cfg['master_zone']
-    except KeyError:
-        logging.warn("master_zone not found in config file, "
-                     "so DNS discovery of master not possible, "
-                     "falling back to AWS EC2 API.")
-        return get_master_ip()
-    if not hasattr(env, 'tag'):
-        env.tag = 'active'
-    try:
-        master = 'master.{0}.{1}.{2}.{3}'.format(env.tag,
-                                                 env.environment,
-                                                 env.application,
-                                                 zone_name)
-        dns.resolver.query(master, 'A')
-    except dns.resolver.NXDOMAIN:
-        master = 'master.{0}.{1}.{2}'.format(env.environment,
-                                             env.application,
-                                             zone_name)
-    return master
-
-
-def set_master_dns():
-    master_ip = get_master_ip()
-    project_config = config.ProjectConfig(env.config,
-                                          env.environment,
-                                          env.stack_passwords)
-    cfg = project_config.config
-    try:
-        zone_name = cfg['master_zone']
-    except KeyError:
-        logging.warn("master_zone not found in config file, "
-                     "no DNS entry will be created, you will "
-                     "need AWS access to deploy etc.")
-        return False
-    r53 = get_connection(R53)
-    zone_id = r53.get_hosted_zone_id(zone_name)
-    if hasattr(env, 'tag'):
-        dns_name = 'master.{0}.{1}.{2}.{3}'.format(env.tag,
-                                                   env.environment,
-                                                   env.application,
-                                                   zone_name)
-    else:
-        dns_name = 'master.{0}.{1}.{2}'.format(env.environment,
-                                               env.application,
-                                               zone_name)
-    r53.update_dns_record(zone_id, dns_name, 'A', master_ip)
-    return True
 
+def get_instance_ips():
+    """
+    Get a list of the public IPs of the current instances in the stack.
+    """
+    cf = get_connection(Cloudformation)
+    ec2 = get_connection(EC2)
+    stack_name = get_stack_name()
+    instance_ids = cf.get_stack_instances(stack_name)
+    instance_ids = [i.id for i in instance_ids]
+    instance_ips = ec2.get_instance_public_ips(instance_ids)
+    return instance_ips
```

Comment:
Can we move this into our EC2 class so that we can write tests for it?

We've also got most of the code for it already: `self.get_instance_public_ips(self.cfn.get_stack_instance_ids(stack_id)) `

(I'm not quite sure why get_instance_public_ips takes a list of instance_ids, but it does)